src/py/flwr/server/superlink/fleet/vce/vce_api.py

```diff
@@ -42,21 +46,143 @@ def _register_nodes(
     return nodes_mapping
 
 
-# pylint: disable=too-many-arguments,unused-argument
+# pylint: disable=too-many-arguments,too-many-locals
+async def worker(
+    app: Callable[[], ClientApp],
+    queue: "asyncio.Queue[TaskIns]",
+    node_states: Dict[int, NodeState],
+    state_factory: StateFactory,
+    nodes_mapping: NodeToPartitionMapping,
+    backend: Backend,
+) -> None:
+    """Get TaskIns from queue and pass it to an actor in the pool to execute it."""
+    state = state_factory.state()
+    while True:
+        try:
+            task_ins: TaskIns = await queue.get()
+            node_id = task_ins.task.consumer.node_id
+
+            # Register and retrive runstate
+            node_states[node_id].register_context(run_id=task_ins.run_id)
+            context = node_states[node_id].retrieve_context(run_id=task_ins.run_id)
+
+            # Convert TaskIns to Message
+            message = message_from_taskins(task_ins)
+            # Replace node-id with data partition id
+            message.metadata.dst_node_id = nodes_mapping[node_id]
```

Comment:
Can we add the partition ID either as an additional metadata field, or as a config value?